static/js/reducers/drawer.js

```diff
@@ -3,19 +3,37 @@ import { createAction, createReducer } from "@reduxjs/toolkit"
 
 export const setDrawerState = createAction("SET_DRAWER_STATE")
 export const setDrawerOpen = createAction("SET_DRAWER_OPEN")
-export const setDrawerMeta = createAction("SET_DRAWER_META")
+export const openDrawer = createAction("OPEN_DRAWER")
+
+type DrawerState = {
+  drawerState: ?string,
+  drawerOpen: boolean,
+  drawerMeta: Object
+}
+
+export type DrawerChangePayload = {
+  type: ?string,
+  meta?: Object
+}
 
 export const drawer = createReducer(
-  { drawerState: null, drawerOpen: false },
+  { drawerState: null, drawerOpen: false, drawerMeta: {} },
   {
-    [setDrawerState]: (state, action) => {
+    [setDrawerState]: (state: DrawerState, action: { payload: ?string }) => {
       state.drawerState = action.payload
     },
-    [setDrawerOpen]: (state, action) => {
+    [setDrawerOpen]: (state: DrawerState, action: { payload: boolean }) => {
       state.drawerOpen = action.payload
     },
-    [setDrawerMeta]: (state, action) => {
-      state.drawerMeta = action.payload
+    [openDrawer]: (
+      state: DrawerState,
+      action: { payload: DrawerChangePayload }
+    ) => {
+      state.drawerState = action.payload.type
+      if (action.payload.meta) {
+        state.drawerMeta = action.payload.meta
+      }
```

Comment:
I think you probably want to reset `drawerMeta` to `{}` if its not passed so you don't accidentally leak state from a previous drawer open into another one.

Reply:
Great catch